### PR TITLE
Updated development dependancies

### LINKED ISSUE #17: Many dependencies are now deprecated, is there any plan to update the latest libraries?
I just clone the master branch and did the installation, here are the command output:
```bash
pi@raspberrypi ~ $ date
Wed Dec  2 23:50:38 SGT 2015

pi@raspberrypi ~/lirc $ git clone https://github.com/alexbain/lirc_web
Cloning into 'lirc_web'...
remote: Counting objects: 475, done.
remote: Total 475 (delta 0), reused 0 (delta 0), pack-reused 475
Receiving objects: 100% (475/475), 636.94 KiB | 229.00 KiB/s, done.
Resolving deltas: 100% (187/187), done.
Checking connectivity... done.

pi@raspberrypi ~/lirc $ cd lirc_web/

pi@raspberrypi ~/lirc/lirc_web $ git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean

pi@raspberrypi ~/lirc/lirc_web $ npm install
npm WARN deprecated swig@0.13.5: v1.0.0 is a complete rewrite of Swig from the ground up. Previous versions are no longer supported
npm WARN deprecated lodash@2.4.2: lodash@<3.0.0 is no longer maintained. Upgrade to lodash@^3.0.0.
npm WARN deprecated lodash@0.9.2: lodash@<3.0.0 is no longer maintained. Upgrade to lodash@^3.0.0.
npm WARN deprecated lodash@1.0.2: lodash@<3.0.0 is no longer maintained. Upgrade to lodash@^3.0.0.
npm WARN optional dep failed, continuing fsevents@1.0.5
consolidate@0.11.0 node_modules/consolidate

grunt-develop@0.3.0 node_modules/grunt-develop

lirc_node@0.0.2 node_modules/lirc_node

swig@0.13.5 node_modules/swig
└── underscore@1.8.3

should@7.1.0 node_modules/should
├── should-type@0.2.0
├── should-equal@0.5.0
└── should-format@0.3.0

sinon@1.17.2 node_modules/sinon
├── formatio@1.1.1
├── samsam@1.1.2
├── lolex@1.3.2
└── util@0.10.3 (inherits@2.0.1)

grunt-contrib-watch@0.6.1 node_modules/grunt-contrib-watch
├── async@0.2.10
├── lodash@2.4.2
├── gaze@0.5.2 (globule@0.1.0)
└── tiny-lr-fork@0.0.5 (debug@0.7.4, qs@0.5.6, faye-websocket@0.4.4, noptify@0.0.3)

supertest@1.1.0 node_modules/supertest
├── methods@1.1.1
└── superagent@1.3.0 (extend@1.2.1, cookiejar@2.0.1, methods@1.0.1, reduce-component@1.0.1, component-emitter@1.1.2, mime@1.3.4, qs@2.3.3, debug@2.2.0, readable-stream@1.0.27-1, formidable@1.0.14, form-data@0.2.0)

mocha@2.3.4 node_modules/mocha
├── escape-string-regexp@1.0.2
├── commander@2.3.0
├── diff@1.4.0
├── growl@1.8.1
├── supports-color@1.2.0
├── debug@2.2.0 (ms@0.7.1)
├── mkdirp@0.5.0 (minimist@0.0.8)
├── glob@3.2.3 (inherits@2.0.1, graceful-fs@2.0.3, minimatch@0.2.14)
└── jade@0.26.3 (commander@0.6.1, mkdirp@0.3.0)

grunt@0.4.5 node_modules/grunt
├── eventemitter2@0.4.14
├── dateformat@1.0.2-1.2.3
├── which@1.0.9
├── getobject@0.1.0
├── async@0.1.22
├── colors@0.6.2
├── lodash@0.9.2
├── rimraf@2.2.8
├── hooker@0.2.3
├── grunt-legacy-util@0.2.0
├── exit@0.1.2
├── coffee-script@1.3.3
├── iconv-lite@0.2.11
├── underscore.string@2.2.1
├── nopt@1.0.10 (abbrev@1.0.7)
├── minimatch@0.2.14 (sigmund@1.0.1, lru-cache@2.7.3)
├── glob@3.1.21 (inherits@1.0.2, graceful-fs@1.2.3)
├── grunt-legacy-log@0.1.2 (grunt-legacy-log-utils@0.1.1, lodash@2.4.2, underscore.string@2.3.3)
├── findup-sync@0.1.3 (lodash@2.4.2, glob@3.2.11)
└── js-yaml@2.0.5 (esprima@1.0.4, argparse@0.1.16)

nodemon@1.8.1 node_modules/nodemon
├── undefsafe@0.0.3
├── es6-promise@3.0.2
├── debug@2.2.0 (ms@0.7.1)
├── touch@1.0.0 (nopt@1.0.10)
├── minimatch@3.0.0 (brace-expansion@1.1.2)
├── lodash.defaults@3.1.2 (lodash.restparam@3.6.1, lodash.assign@3.2.0)
├── ps-tree@1.0.1 (event-stream@3.3.2)
├── update-notifier@0.5.0 (is-npm@1.0.0, chalk@1.1.1, string-length@1.0.1, semver-diff@2.1.0, repeating@1.1.3, configstore@1.3.0, latest-version@1.0.1)
└── chokidar@1.3.0 (path-is-absolute@1.0.0, async-each@0.1.6, glob-parent@2.0.0, is-binary-path@1.0.1, is-glob@2.0.1, readdirp@2.0.0, anymatch@1.3.0)

grunt-contrib-uglify@0.9.2 node_modules/grunt-contrib-uglify
├── uri-path@0.0.2
├── chalk@1.1.1 (escape-string-regexp@1.0.3, ansi-styles@2.1.0, supports-color@2.0.0, strip-ansi@3.0.0, has-ansi@2.0.0)
├── uglify-js@2.6.1 (async@0.2.10, uglify-to-browserify@1.0.2, source-map@0.5.3, yargs@3.10.0)
├── maxmin@1.1.0 (figures@1.4.0, gzip-size@1.0.0, pretty-bytes@1.0.4)
└── lodash@3.10.1

jsdom@7.1.0 node_modules/jsdom
├── acorn-globals@1.0.9
├── webidl-conversions@2.0.1
├── abab@1.0.1
├── xml-name-validator@2.0.1
├── symbol-tree@3.1.4
├── tough-cookie@2.2.1
├── nwmatcher@1.3.7
├── cssom@0.3.0
├── whatwg-url-compat@0.6.5 (tr46@0.0.2)
├── parse5@1.5.1
├── acorn@2.6.4
├── htmlparser2@3.9.0 (domelementtype@1.3.0, entities@1.1.1, domhandler@2.3.0, readable-stream@2.0.4, domutils@1.5.1)
├── escodegen@1.7.1 (estraverse@1.9.3, esutils@2.0.2, esprima@1.2.5, optionator@0.5.0, source-map@0.2.0)
├── cssstyle@0.2.30
└── request@2.67.0 (aws-sign2@0.6.0, forever-agent@0.6.1, tunnel-agent@0.4.1, oauth-sign@0.8.0, is-typedarray@1.0.0, caseless@0.11.0, stringstream@0.0.5, isstream@0.1.2, json-stringify-safe@5.0.1, extend@3.0.0, node-uuid@1.4.7, qs@5.2.0, mime-types@2.1.8, combined-stream@1.0.5, form-data@1.0.0-rc3, bl@1.0.0, hawk@3.1.2, http-signature@1.1.0, har-validator@2.0.3)

express@3.0.6 node_modules/express
├── methods@0.0.1
├── fresh@0.1.0
├── cookie-signature@0.0.1
├── range-parser@0.0.4
├── buffer-crc32@0.1.1
├── cookie@0.0.5
├── commander@0.6.1
├── mkdirp@0.3.3
├── send@0.1.0 (mime@1.2.6)
├── debug@2.2.0 (ms@0.7.1)
└── connect@2.7.2 (pause@0.0.1, bytes@0.1.0, qs@0.5.1, formidable@1.0.11)

grunt-contrib-less@1.0.1 node_modules/grunt-contrib-less
├── async@0.9.2
├── lodash@2.4.2
├── chalk@0.5.1 (ansi-styles@1.1.0, escape-string-regexp@1.0.3, supports-color@0.2.0, strip-ansi@0.3.0, has-ansi@0.1.0)
└── less@2.4.0 (mime@1.3.4, graceful-fs@3.0.8, image-size@0.3.5, errno@0.1.4, promise@6.1.0, mkdirp@0.5.1, source-map@0.2.0, request@2.67.0)

pi@raspberrypi ~/lirc/lirc_web $ node app.js
DEBUG: { [Error: Cannot find module '/home/pi/lirc/lirc_web/config.json'] code: 'MODULE_NOT_FOUND' }
WARNING: Cannot find config.json!
Open Source Universal Remote UI + API has started on port 3000.
```

The app is still running perfectly fine. I can control my fan without any issue from the web interface.
If there's no plan to update the dependencies, feel free to close this issue.

### PR DESCRIPTION
**Pull request by pkubat**
Updated development dependencies, this closed #17.  Note that lodash 4.0.0 was just released yesterday, I did not include that update.

### CODE REVIEW

**Comment by pkubat** [comment]
It looks like travis is using really old versions of node and npm?  Might be related to the failed checks.

File: .eslintrc [hunk 1]
```diff
@@ -3,7 +3,8 @@
   "rules": {
     "func-names": 0,
     "no-console": 0,
-    "guard-for-in": 0
+    "guard-for-in": 0,
+    "max-len": 1
```
Comment (pkubat) [inline .eslintrc]: Changed to a warning.

**Comment by pkubat** [comment]
thats better, passing!

Now using node v5.4.1
$ node --version
v5.4.1
$ npm --version
3.3.12
$ nvm --version
0.23.3

**Comment by alexbain** [comment]
Thanks!

**Pull request merged**